http/go/nethttp/kiota_client_factory.go

```diff
@@ -27,6 +27,7 @@ func GetDefaultMiddlewares() []Middleware {
 	return []Middleware{
 		NewRetryHandler(),
 		NewRedirectHandler(),
+		&CompressionHandler{},
```

Comment:
- add an option struct so people can disable the compression on a single request should they want to
- add a default constructor for the compression handler
- add an overload constructor that accepts the options for the compression handler
(see redirect handler)

Reply:
> add an option struct so people can disable the compression on a single request should they want to

This requires a change to the [spec](https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/middleware/CompressionHandler.md)

Reply:
good catch, I forgot to mention it in my original spec, [created a PR](https://github.com/microsoftgraph/msgraph-sdk-design/pull/74)